### ISSUE TITLE
Dropdown offset seems strange...

### ISSUE THREAD

**Issue by DavidCRivera** [Dogfalo/materialize#2039]
Not sure if this is a bug in the scripting itself or in my implementation, but I'd appreciate the help either way. I've created a fiddle of the basic layout of this part of my page. Ever since upgrading to 0.97.1, the dropdowns appear in weird places. They seem to be close to where they should  be when the viewport is small and the columns containing them are s12, but when it goes up to m6 or something similar, the position is way left of where it should be. Any idea what the issue is? 

http://jsfiddle.net/5LawL4ez/2/

Thanks!

**Comment by Dogfalo** [comment]
Seems to be a problem related to the right alignment. We will look into this

**Comment by msolters** [comment]
I'm checking this out right now.  Part of it is the fact that, while we are checking for the dropdown potentially going off-screen (beyond the bounds of the window), we aren't checking if an alignment is forcing the dropdown list outside the bounds of its parent element.  So, if something is not visible to the window we change its alignment, but if something is not visible to the parent we just let it disappear.

**Comment by msolters** [comment]
How does [this work for you](https://jsfiddle.net/5LawL4ez/2/)?  I adjusted the alignment calculations to not rely on offsets or `$(window).width()`, and instead to be do absolute position by `origin.position()`.  It seems to work as expected.

```js
      // Offscreen detection
      var offsetLeft = origin.offset().left;
      var leftPosition, gutter_spacing;
      if (offsetLeft + activates.innerWidth() > $(window).width()) {
      	//console.log("Forcing right alignnment.");
        options.alignment = 'right';
      }
      else if (offsetLeft - activates.innerWidth() + origin.outerWidth() < 0) {
        //console.log("Forcing left alignment.");
        options.alignment = 'left';
      }

      // Handle edge alignment
      if (options.alignment === 'left') {
        gutter_spacing = options.gutter;
        leftPosition = origin.position().left + gutter_spacing;
      }
      else if (options.alignment === 'right') {
        var offsetRight = origin.position().left + origin.outerWidth() - activates.outerWidth();
        gutter_spacing = -options.gutter;
        leftPosition =  offsetRight + gutter_spacing;
      }
      // Position dropdown
      activates.css({
        position: 'absolute',
        top: origin.position().top + offset,
        left: leftPosition
      });
```

**Comment by msolters** [comment]
Please note that I just made 3 edits to the above post, so if you visited the JSFiddle from an e-mail notification you probably saw an out-of-date attempt that doesn't work.

**Comment by DavidCRivera** [comment]
I appreciate the quick response/fix. Looking forward to the next release. Thanks!

**Comment by msolters** [comment]
Hey @DavidCRivera, so the above jsfiddle link is working fine for you?

**Comment by acburst** [comment]
Fixed in 561ee50
